docs/7-push-notifications.md

```diff
@@ -0,0 +1,245 @@
+
+```xml
+<dict>
+<key>UIBackgroundModes</key>
+    <array>
+        <string>remote-notification</string>
+    </array>
+```
+
+Alternatively, you can do the following: **Targets** -> **Your App** -> **Capabilities** -> **Background Modes** and check Remote notifications. 
+
+![Remotenotifications.png](./Remotenotifications.png)
+
+Given the way the iOS Push Notification work, it’s critical for the user to grant your app permission on the first pass. If the user taps “Don’t Allow”, it takes multiple taps within the Settings app to re-grant permission, making it extremely unlikely, and practically irreversible so many developers layer a recovable request prior to displaying Apple's irrevocable one. 
+
+### 4.1 Registering for Remote Notifications
+
+As soon as the `Info.plist` is configured, we have to register for remote notifications with Apple's method called `registerForRemoteNotifications`. Since Push Notifications run disrupt an app's User Interface, registration needs to happen on the main thread. To register on the main thread we pass `registerForRemoteNotifications` as an argument in the `async().` method on `.main` property for `DispatchQueue`:
+
+```swift
+DispatchQueue.main.async(execute: UIApplication.shared.registerForRemoteNotifications)
+```
+### 4.2 Registering Device Token with the SDK
+
+After the functionality for registering remote notifications is integrated, we program `.didRegisterForRemoteNotificationsWithDeviceToken` to pass the device token to the SDK: 
+
+```swift
+// MARK:
+// MARK: Notification
+
+func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+client.appLifecycle.push.registeredForRemoteNotifications(with: deviceToken)
+}
+```
+
+## 5 Configuring the Stitch iOS SDK within the iOS App 
+
+The last step is to configure the app to respond to requests for Push Notifications. Drop either of the following lines of code inside of `viewDidLoad(:)` in `ViewController.swift`:
+
+```swift
+// listen for push notifications
+ConversationClient.instance.appLifecycle.notifications.subscribe(onSuccess: { notification in
+// code here
+})
+
+```
+You can also subscribe to events for push notifications through the `.receiveRemoteNotifications` property here: 
+
+```
+ConversationClient.instance.appLifecycle.receiveRemoteNotification.subscribe(onSuccess: { notification in
+// code here
+})
+```
+
+## 5.1 Configuring `sendBtn`
+
+In `ViewController.swift` drop in an instance of `ConversationClient` so that we can send messages. 
+
+```
+ /// Nexmo Conversation client
+    let client: ConversationClient = {
+        return ConversationClient.instance
+    }()
+```
+Next we configure our `sendBtn` to send messages through our `ConversationClient`:
```

Comment:
you forgot two major steps requestion push notification permission and app delegate stuff.

[request permission](https://developer.nexmo.com/sdk/stitch/ios/Classes/PushNotificationController.html#/c:@M@Stitch@objc(cs)NXMPushNotificationController(im)requestPermission)

[appdelegate method](https://developer.nexmo.com/sdk/stitch/ios/Classes/PushNotificationController.html#/s:6Stitch26PushNotificationControllerC014receivedRemoteC0XeXeF)

Reply:
@shams-ahmed Thanks for adding the references.